kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,897 @@
+     * @details This function adds a linear solver to the collection without extending parameters.
+     * @param pSolver Pointer to the linear solver to be added.
+     */
+    void AddSolver(LinearSolverPointer pSolver)
+    {
+        // Increase the solvers vector
+        mSolvers.push_back(pSolver);
+    }
+
+    /**
+     * @brief Add a linear solver to the collection with additional parameters.
+     * @details This function adds a linear solver to the collection and extends the parameters.
+     * @param pSolver Pointer to the linear solver to be added.
+     * @param ThisParameters Parameters associated with the linear solver.
+     * @note This function extends parameters, but the interface for this is yet to be decided.
+     */
+    void AddSolver(
+        LinearSolverPointer pSolver,
+        const Parameters ThisParameters
+        )
+    {
+        // Increase the solvers vector
+        AddSolver(pSolver);
+
+        // Extend the parameters
+        FillParametersFromSolver(pSolver);
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    /**
+     * @brief Get the Solvers list.
+     * @return const std::vector<LinearSolverPointer>& Reference to the vector of solvers.
+     */
+    const std::vector<LinearSolverPointer>& GetSolvers() const
+    {
+        return mSolvers;
+    }
+
+    /**
+     * @brief Set the Solvers list.
+     * @param rSolvers A vector of LinearSolverPointer to set.
+     */
+    void SetSolvers(const std::vector<LinearSolverPointer>& rSolvers)
+    {
+        mSolvers = rSolvers;
+    }
+
+    /**
+     * @brief Check if the solver index should be reset for each try.
+     * @return true If the solver index is reset for each try.
+     * @return false Otherwise.
+     */
+    bool& GetResetSolverIndexEachTry()
+    {
+        return mResetSolverIndexEachTry;
+    }
+
+    /**
+     * @brief Set whether the solver index should be reset for each try.
+     * @param Reset Flag indicating whether to reset the solver index each try.
+     */
+    void SetResetSolverIndexEachTry(const bool Reset)
+    {
+        mResetSolverIndexEachTry = Reset;
+    }
+
+    /**
+     * @brief Get the Parameters.
+     * @return Parameters copy to the Parameters object.
+     */
+    Parameters GetParameters() const
+    {
+        return mParameters;
+    }
+
+    /**
+     * @brief Set the Parameters.
+     * @param parameters A Parameters object to set.
+     */
+    void SetParameters(const Parameters ThisParameters)
+    {
+        mParameters = ThisParameters;
+    }
+
+    /**
+     * @brief Get the Current Solver Index.
+     * @return IndexType The current solver index.
+     */
+    IndexType& GetCurrentSolverIndex()
+    {
+        return mCurrentSolverIndex;
+    }
+
+    /**
+     * @brief Set the Current Solver Index.
+     * @param index The new solver index to set.
+     */
+    void SetCurrentSolverIndex(const IndexType Index)
+    {
+        mCurrentSolverIndex = Index;
+    }
```

Comment:
`GetCurrentSolverIndex` returns by mutable reference (very bad idea) so `SetCurrentSolverIndex` is unnecessary. I'd just remove these functions too and maybe expose a function that resets the counter.